src/Configuration.php

```diff
@@ -1823,4 +1823,34 @@ private static function looksLikeAPipe($stream): bool
 
         return $mode === 0010000 || $mode === 0040000 || $mode === 0100000 || $mode === 0120000;
     }
+
+    /**
+     * Locate a command (an executable) in $PATH.
+     *
+     * Behaves like 'command -v COMMAND' or 'which COMMAND'.
+     * If $PATH is unset/empty it defaults to '/usr/sbin:/usr/bin:/sbin:/bin'.
+     *
+     * @param string $command the executable to locate
+     *
+     * @return string
+     */
+    private static function which($command)
+    {
+        $path = \getenv('PATH');
+        if (empty($path)) {
+            $path = '/usr/sbin:/usr/bin:/sbin:/bin';
+        }
+        $paths = \explode(':', $path);
+        foreach ($paths as $path) {
+            $fullpath = "$path/$command";
+            $st = @\stat($fullpath);
+            if ($st !== false &&
+                    ($st['mode'] & 0170000) === 0100000 &&   // S_IFREG
```

Comment:
should we also include symlinks?

Reply:
`stat()` follows symlinks (we didn't use `lstat()`), so you would never encounter a symlink:
```
$ ls -l /bin/sh
lrwxrwxrwx 1 root root 4 mrt 23 14:49 /bin/sh -> dash
```
```
$ python3 -c 'import os; s=os.stat("/bin/sh"); print(oct(s.st_mode))'
0o100755
```
vs.
```
$ python3 -c 'import os; s=os.lstat("/bin/sh"); print(oct(s.st_mode))'
0o120777
```